client/src/main/java/org/eclipse/hono/client/MessageSender.java

```diff
@@ -27,70 +27,30 @@
  * A client for publishing messages to Hono.
  *
  */
-public interface MessageSender {
+public interface MessageSender extends CreditBasedSender{
 
     /**
      * Gets the name of the endpoint this sender sends messages to.
      * <p>
      * The name returned is implementation specific, e.g. an implementation
      * that can be used to upload telemetry data to Hono will return
      * the value <em>telemetry</em>.
-     * 
+     *
      * @return The endpoint name.
      */
     String getEndpoint();
 
-    /**
-     * Gets the number of messages this sender can send based on its current number of credits.
-     * <p>
-     * Note that the value returned is valid during execution of the current vert.x handler only.
-     * 
-     * @return The number of messages.
-     */
-    int getCredit();
-
     /**
      * Checks if this sender can send or buffer (and send later) a message.
      * <p>
-     * Note that the value returned is valid during execution of the current vert.x handler only.
+     * Note that the value returned is valid durin execution of the current vert.x handler only.
```

Comment:
during

Reply:
Thanks for noticing it. It has been corrected.